js-w3-hw.js

```diff
@@ -0,0 +1,144 @@
+  "Peter",
+  "Ahmad",
+  "Yana",
+  "kristina",
+  "Rasmus",
+  "Samuel",
+  "katrine",
+  "Tala",
+];
+
+const nameToRemove = "Ahmad";
+const index = names.indexOf(nameToRemove); // Index of the name to remove
+
+if (index !== -1) {
+  // Check if the name is found, and then remove it
+  names.splice(index, 1);
+}
+
+console.log(names); // Log out the remaining array
+
+// EX2. When will we be there??
+
+function calculateTravelTime(travelInformation) {
+  const { speed, destinationDistance } = travelInformation;
+
+  // Calculate the time in hours as a decimal
+  const timeInHours = destinationDistance / speed;
+
+  // Extract the whole number of hours
+  const hours = Math.floor(timeInHours);
+
+  // Calculate the remaining minutes and round it to the nearest integer.
+  const minutes = Math.round((timeInHours - hours) * 60);
+
+  // Return the formatted result
+  return `${hours} hours and ${minutes} minutes`;
+}
+
+// Example
+const travelInformation = {
+  speed: 50,
+  destinationDistance: 432,
+};
+
+const travelTime = calculateTravelTime(travelInformation);
+console.log(travelTime); // "8 hours and 38 minutes"
+
+// EX3. Series duration of my life
+
+const seriesDurations = [
+  {
+    title: "Friends",
+    days: 4,
+    hours: 4,
+    minutes: 36,
+  },
+  {
+    title: "The Office",
+    days: 3,
+    hours: 19,
+    minutes: 24,
+  },
+  {
+    title: "High Maintenance",
+    days: 0,
+    hours: 17,
+    minutes: 0,
+  },
+];
+
+function logOutSeriesText() {
+  let totalPercentage = 0; // initialization to 0 before the loop starts. This variable will hold the cumulative percentage.
+  
+  const minutesInYear = 365.25 * 24 * 60; // Calculates minutes in a year accounting for leap years
+  const averageLifespanMinutes = 80 * minutesInYear; // Calculates average 80 years in minutes
+
+// The code below starts a loop using the forEach method. forEach takes a callback function and executes it once for each array element, with 'series' representing the current object.
+  seriesDurations.forEach((series) => {
+    const { days, hours, minutes } = series;
+    const totalMinutes = (days * 24 * 60) + (hours * 60) + minutes; // Converts the duration of the series from days, hours, and minutes to the total number of minutes.
+    const percentageOfLife = (totalMinutes / averageLifespanMinutes) * 100; // Converts the fraction of life spent watching the series into a percentage.
+
+    totalPercentage += percentageOfLife; // Accumulate the percentage of life spent watching each series into totalPercentage
+
+// Each series percentage is logged, followed by the total percentage.
+    console.log(
+      `${series.title} took ${percentageOfLife.toFixed(3)}% of my life`
+    );
+  });
+
+  console.log(`In total that is ${totalPercentage.toFixed(3)}% of my life`);
+}
+
+logOutSeriesText();
+
+// EX4. NOnoN0nOYes (Note taking app)
+
+const notes = []; // Array initialization
+
+function saveNote(content, id) {
+  // Function for saving a note
```

Comment:
YOu've done a very good job adding comments here and there. It's a good habit and helpful both for you and anyone else looking at the code. Just try to balance where you add them, as sometimes the names of variables and functions can be self-explanatory and make comments redundant or simply repetitive. This comment here can pass for one of the "one too many".